src/main/java/marquez/api/models/DatasetResponse.java

```diff
@@ -0,0 +1,31 @@
+package marquez.api.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Optional;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+
+@RequiredArgsConstructor
+@EqualsAndHashCode
+@ToString
+public class DatasetResponse {
```

Comment:
I see that we are able to get away with not having a `NoArgsConstructor`. This makes me hopeful we can remove it elsewhere.